include/swift/AST/DiagnosticsSema.def

```diff
@@ -4699,6 +4699,11 @@ ERROR(actor_isolation_superclass_mismatch,none,
       "%0 class %1 has different actor isolation from %2 superclass %3",
       (ActorIsolation, DeclName, ActorIsolation, DeclName))
 
+ERROR(warn_async_function_must_be_available_from_async,none,
+        "asynchronous function %0 must be available from asynchronous contexts", (DeclName))
+WARNING(warn_async_unavailable_decl,none,
```

Comment:
Do we think this should be an error in Swift 6 mode, or will it always be a warning? Or should there be more of a distinction for the user about, e.g., "deprecated" (warning) vs. "unavailable" (error, albeit maybe only in Swift 6)?